pkg/discovery/discovery.go

```diff
@@ -143,7 +143,7 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 		// In some failure situations, for example, delete the pd's data directory, pd will try to restart
 		// and get join info from discovery service. But pd embed etcd may still have the registered member info,
 		// which will return the argument to join pd itself, which is not suggested in pd.
-		if member.Name == podName {
+		if member.Name == podName || member.Name == strArr[0] {
```

Comment:
Can you give an example in the comment here?

Reply:
It corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).

Reply:
I mean, you can add this comment in the code (for other developers or even you). 😹

Reply:
We should leave comments as is done in the kubernetes project...